subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -339,6 +339,19 @@ Gradle changes the compression library used for creating archives from an Ant ba
 The embedded Kotlin has been updated to https://github.com/JetBrains/kotlin/releases/tag/v1.7.22[Kotlin 1.7.22].
 For more information, see the release notes for Kotlin https://github.com/JetBrains/kotlin/releases/tag/v1.7.20[1.7.20] and https://github.com/JetBrains/kotlin/releases/tag/v1.7.21[1.7.21].
 
+[[kotlin_language_1_7]]
+==== Kotlin DSL now uses Kotlin language/api 1.7
+
+Starting with Gradle 8.0 it uses Kotlin language and API 1.7 level.
+It was previously using 1.4.
+This includes potential breaking changes for your `.gradle.kts` scripts and plugins authored with the `kotlin-dsl` plugin.
+
+Please find below links to the included changes:
+
+* Kotlin 1.5 link:https://kotlinlang.org/docs/whatsnew15.html#language-features[language] / link:https://kotlinlang.org/docs/whatsnew17.html#standard-library[stdlib]
+* Kotlin 1.6 link:https://kotlinlang.org/docs/whatsnew16.html#language[language] / link:https://kotlinlang.org/docs/whatsnew16.html#standard-library[stdlib]
+* Kotlin 1.7 link:https://kotlinlang.org/docs/whatsnew17.html#language[language] / link:https://kotlinlang.org/docs/whatsnew15.html#standard-library[stdlib]
+
```

Comment:
```suggestion
[[kotlin_language_1_7]]
==== Updated the Kotlin DSL to Kotlin 1.7

Previously, the Kotlin DSL used Kotlin 1.4.
Starting with Gradle 8.0, the Kotlin DSL uses Kotlin 1.7.
This change brings all the improvements made to the Kotlin language and standard library since Kotlin 1.4.0.

The following links explain the changes included in this upgrade:

* Kotlin 1.5 link:https://kotlinlang.org/docs/whatsnew15.html#language-features[language] / link:https://kotlinlang.org/docs/whatsnew17.html#standard-library[stdlib]
* Kotlin 1.6 link:https://kotlinlang.org/docs/whatsnew16.html#language[language] / link:https://kotlinlang.org/docs/whatsnew16.html#standard-library[stdlib]
* Kotlin 1.7 link:https://kotlinlang.org/docs/whatsnew17.html#language[language] / link:https://kotlinlang.org/docs/whatsnew15.html#standard-library[stdlib]

```
Suggestion: similar changes to those made above.
Since this isn't an actionable item for most users, consider removing this section from the upgrading guide. If it is actionable, we should specifically call out the users who ought to react to this and provide suggestions for how to react (maybe users should migrate to exhaustive `when`, for instance?).

Reply:
We can't really list all possible changes, this is the responsibility of the Kotlin documentation. I don't think we need to go into details here. We must have some words about this here because this includes potential breaking changes.

Reply:
That's fine. Perhaps we can mention that responsibility explicitly -- "For information about breaking and nonbreaking changes resulting from this upgrade, see the following links to the Kotlin documentation".